elrond-wasm-derive/src/parse/method_parse.rs

```diff
@@ -1,27 +1,110 @@
-use crate::model::Method;
+use crate::model::{Method, MethodImpl, MethodPayableMetadata, PublicRole};
 
 use super::{
-	attributes::{extract_doc, find_output_names},
-	extract_method_args,
-	method_impl_parse::process_method_impl,
-	process_public_role,
+	attributes::extract_doc,
+	auto_impl_parse::{
+		process_event_attribute, process_legacy_event_attribute, process_proxy_attribute,
+		process_storage_clear_attribute, process_storage_get_attribute,
+		process_storage_is_empty_attribute, process_storage_mapper_attribute,
+		process_storage_set_attribute,
+	},
+	extract_method_args, process_callback_attribute, process_callback_raw_attribute,
+	process_endpoint_attribute, process_init_attribute, process_output_names_attribute,
+	process_payable_attribute, process_view_attribute,
 };
+pub struct MethodAttributesPass1 {
+	pub method_name: String,
+	pub payable: MethodPayableMetadata,
+}
 
 pub fn process_method(m: &syn::TraitItemMethod) -> Method {
-	let public_role = process_public_role(m);
 	let method_args = extract_method_args(m);
-	let output_names = find_output_names(m);
-	let implementation = process_method_impl(m);
 
-	Method {
+	let implementation = if let Some(body) = m.default.clone() {
+		MethodImpl::Explicit(body)
+	} else {
+		MethodImpl::NoImplementation
+	};
+
+	let mut pass_1_data = MethodAttributesPass1 {
+		method_name: m.sig.ident.to_string(),
+		payable: MethodPayableMetadata::NotPayable,
+	};
+	let mut pass_1_unprocessed_attributes = Vec::new();
+
+	process_attributes_pass_1(
+		&m.attrs,
+		&mut pass_1_data,
+		&mut pass_1_unprocessed_attributes,
+	);
+
+	let mut method = Method {
 		docs: extract_doc(m.attrs.as_slice()),
-		public_role,
+		public_role: PublicRole::Private,
 		name: m.sig.ident.clone(),
 		generics: m.sig.generics.clone(),
-		remaining_attributes: Vec::new(), // TODO: keep unprocessed attributes (e.g. `#[inline]` here)
+		unprocessed_attributes: Vec::new(),
 		method_args,
-		output_names,
+		output_names: Vec::new(),
 		return_type: m.sig.output.clone(),
 		implementation,
+	};
+
+	process_attributes_pass_2(&pass_1_unprocessed_attributes, &pass_1_data, &mut method);
+
+	method
+}
+
+fn process_attributes_pass_1(
+	attrs: &[syn::Attribute],
+	pass_1_data: &mut MethodAttributesPass1,
+	pass_1_unprocessed_attributes: &mut Vec<syn::Attribute>,
+) {
+	for attr in attrs {
+		let processed = process_attribute_pass_1(attr, pass_1_data);
+		if !processed {
+			pass_1_unprocessed_attributes.push(attr.clone());
+		}
 	}
 }
+
+fn process_attribute_pass_1(
```

Comment:
Not a fan of this naming model for two reasons: it's not clear what pass_1 does different that pass_2 + is also pretty easy to mix them up, having only one character that differs. At least with a similar naming like pass_first and pass_second you can resolve the second issue, although a more meaningful name would be preffered